src/huggingface_hub/snapshot_download.py

```diff
@@ -41,8 +42,17 @@ def snapshot_download(
     if isinstance(cache_dir, Path):
         cache_dir = str(cache_dir)
 
+    if isinstance(use_auth_token, str):
```

Comment:
Here I aim to cover both cases where the user provides an explicit API token or we retrieve it from the cache

Reply:
I agree with this!